library/src/main/java/com/mapzen/android/MapzenMap.java

```diff
@@ -676,4 +708,26 @@ public void drawTransitRouteLine(@NotNull List<LngLat> points, @Nullable List<Ln
   public void clearTransitRouteLine() {
     overlayManager.clearTransitRouteLine();
   }
+
+  /**
+   * By default all {@link MapData} objects are removed from the map when it is destroyed. To
+   * persist this data such as in the case of orientation changes, use this method.
+   * @param persistOnRecreation persist {@link MapData} across orientation changes
+   */
+  public void setPersistMapData(boolean persistOnRecreation) {
```

Comment:
As discussed, I'm on board with this.

Added a separate method `setPersistMapState` and have it defaulting to `true`